tchannel/tornado/peer.py

```diff
@@ -656,6 +677,34 @@ def _update_heap(self, peer):
         peer.rank = rank
         self.peer_heap.update_peer(peer)
 
+    def get_for_request(self, hostport):
+        """Get a Peer for the given destination for a request.
+
+        A new Peer is added and returned if one does not already exist for the
+        given host-port. Otherwise, the existing Peer is returned.
+
+        **NOTE** new peers will not be added to the peer heap.
+        """
+        assert hostport, "hostport is required"
+        if hostport not in self._peers:
+            self.add_direct(hostport)
+
+        return self._peers[hostport]
+
+    def add_direct(self, hostport):
+        """
+        Add a peer directly from a hostport, do NOT add it to the peer heap
+        """
+        peer = self.peer_class(
+            tchannel=self.tchannel,
+            hostport=hostport,
+        )
+
```

Comment:
shooooooooouldn't need it, but paranoia, yay

hmm, this function is pretty small without that, maybe we can just inline it...